x-pack/plugins/viz_editor/common/query_types.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// ----------------------------------------
+// Select clause
+// ----------------------------------------
+
+export interface Aliasable {
+  alias?: string;
+}
```

Comment:
You've said that alias is required in the sanitize file, but here it's optional. I don't understand how you expect `alias` to get used, can you clarify?

Reply:
Yeah. I'm not sure how best to represent this in TypeScript. Basically, as an input, it's optional, but internally to the translation / query layer, it's really handy for it to be required. We can do this in one of two ways:

- Let it be optional as an input, and sanitize our inputs, so that internally, it's always set
- Make it required as an input, and push the work of properly "naming" a column out to whatever code is building the query

I can chat more, if that's not clear.